modules/rostests/apitests/shell32/ShellState.cpp

```diff
@@ -0,0 +1,226 @@
+    printf("\n");
+
+    printf("WINVER: 0x%04X\n", WINVER);
+    printf("_WIN32_WINNT: 0x%04X\n", _WIN32_WINNT);
+    printf("_WIN32_IE: 0x%04X\n", _WIN32_IE);
+    printf("NTDDI_VERSION: 0x%08X\n", NTDDI_VERSION);
+
+#ifdef UNICODE
+    printf("UNICODE\n");
+#else
+    printf("ANSI\n");
+#endif
+
+#ifdef _MSC_VER
+    printf("_MSC_VER: 0x%08X\n", int(_MSC_VER));
+#elif defined(__MINGW32__)
+    printf("__MINGW32__: 0x%08X\n", int(__MINGW32__));
+#elif defined(__clang__)
+    printf("__clang__: 0x%08X\n", int(__clang__));
+#else
+    #error Unknown compiler.
+#endif
+
+    printf("\n");
+
+    ok(sizeof(REGSHELLSTATE) >= 0x24, "sizeof(REGSHELLSTATE) was %d\n", (int)sizeof(REGSHELLSTATE));
+    printf("sizeof(SHELLSTATE): %d\n", (int)sizeof(SHELLSTATE));
+    printf("__alignof(SHELLSTATE): %d\n", (int)__alignof(SHELLSTATE));
+    printf("sizeof(SHELLFLAGSTATE): %d\n", (int)sizeof(SHELLFLAGSTATE));
+    printf("sizeof(CABINETSTATE): %d\n", (int)sizeof(CABINETSTATE));
+
+    pss = &rss.ss;
+    pb = (LPBYTE)pss;
+
+    ret = read_key(&rss);
+    if (ret)
+    {
+        return;
+    }
+
+    dump_pss(pss);
+    ok(rss.dwSize >= 0x24, "rss.dwSize was %ld (0x%lX).\n", rss.dwSize, rss.dwSize);
+
+#define DUMP_LONG(x) printf(#x ": 0x%08X\n", int(x));
+#define DUMP_BOOL(x) printf(#x ": %d\n", !!int(x));
+    DUMP_BOOL(pss->fShowAllObjects);
+    DUMP_BOOL(pss->fShowExtensions);
+    DUMP_BOOL(pss->fNoConfirmRecycle);
+    DUMP_BOOL(pss->fShowSysFiles);
+    DUMP_BOOL(pss->fShowCompColor);
+    DUMP_BOOL(pss->fDoubleClickInWebView);
+    DUMP_BOOL(pss->fDesktopHTML);
+    DUMP_BOOL(pss->fWin95Classic);
+    DUMP_BOOL(pss->fDontPrettyPath);
+    DUMP_BOOL(pss->fShowAttribCol);
+    DUMP_BOOL(pss->fMapNetDrvBtn);
+    DUMP_BOOL(pss->fShowInfoTip);
+    DUMP_BOOL(pss->fHideIcons);
+    DUMP_BOOL(pss->fWebView);
+    DUMP_BOOL(pss->fFilter);
+    DUMP_BOOL(pss->fShowSuperHidden);
+    DUMP_BOOL(pss->fNoNetCrawling);
+    DUMP_LONG(pss->lParamSort);
+    DUMP_LONG(pss->iSortDirection);
+    DUMP_LONG(pss->version);
+    DUMP_LONG(pss->lParamSort);
+    DUMP_LONG(pss->iSortDirection);
+    DUMP_LONG(pss->version);
+    DUMP_BOOL(pss->fSepProcess);
+    DUMP_BOOL(pss->fStartPanelOn);
+    DUMP_BOOL(pss->fShowStartPage);
+#if NTDDI_VERSION >= 0x06000000     // for future use
+    DUMP_BOOL(pss->fIconsOnly);
+    DUMP_BOOL(pss->fShowTypeOverlay);
+    DUMP_BOOL(pss->fShowStatusBar);
+#endif
+
+#define SSF_MASK \
+    (SSF_SHOWALLOBJECTS | SSF_SHOWEXTENSIONS | SSF_NOCONFIRMRECYCLE | \
+     SSF_SHOWCOMPCOLOR | SSF_DOUBLECLICKINWEBVIEW | SSF_DESKTOPHTML | \
+     SSF_WIN95CLASSIC | SSF_DONTPRETTYPATH | SSF_SHOWATTRIBCOL | \
+     SSF_MAPNETDRVBUTTON | SSF_SHOWINFOTIP | SSF_HIDEICONS)
+    // For future:
+    // SSF_AUTOCHECKSELECT, SSF_ICONSONLY, SSF_SHOWTYPEOVERLAY, SSF_SHOWSTATUSBAR
+
+    memset(&FlagState, 0, sizeof(FlagState));
+    SHGetSettings(&FlagState, SSF_MASK);
+
+#define CHECK_FLAG(x) ok(pss->x == FlagState.x, "%s\n", #x)
```

Comment:
Printing the wrong/expected value would be nice.

Reply:
OK.